mod-orgs/schemas/interface_credentials.json

```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "An interface record",
+  "type": "object",
+  "properties": {
+    "id": {
```

Comment:
After another thought having just id might be problematic/confusing:
1. The description of the schema has to be clear enough indicating that this id is the same as interface id
2. In this case primary key has to be the same time also as foreign key to `interfaces` table. Looks like postgres allows such case (at least I do not see any restriction in the documentation) but it might be confusing a bit

Reply:
I see that the JIRA story originally had a separate field `interfaceId` which held a reference to the interface the credentials applied to.  However, the story was changed to remove this. 

We shouldn't be reusing UUIDs to apply to multiple records, regardless of their relationship.  Please implement the schema with a separate interfaceId field.

Reply:
These changes leads to two-phase processing of getCredentialById and deleteCredentialById endpoints:
1) Get List of all interface credentials by query `interfaceId==${id}` and `offset=0` and `limit=1`
2) Depends on endpoint:
#### GET /organizations-storage/interfaces/{id}/credentials
- Return first element. `PgUtils` can't be used, all error cases should be handled manually.
#### DELETE /organizations-storage/interfaces/{id}/credentials
- Invoke delete request by id of the first element, return empty `response204`. PgUtils.deleteById can be used. Anyway, all error cases also should be handled manually.

Please confirm this approach.

Reply:
If PgUtils doesn't support what we're trying to do here than it's OK to not use it.

In theory you should be able to:

GET by id:
`SELECT jsonb FROM credentials WHERE jsonb->>'interfaceId'='${id}'`

DELETE by id:
`DELETE FROM credentials WHERE jsonb->>'interfaceId'='${id}'`

Also, it probably makes sense to add a unique index to the interfaceId field so you don't have to worry about getting back multiple records

Reply:
Schema and [Implementation](https://github.com/folio-org/mod-organizations-storage/pull/22) updated